pkg/apis/machine/v1beta1/machine_webhook_test.go

```diff
@@ -0,0 +1,234 @@
+package v1beta1
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	osconfigv1 "github.com/openshift/api/config/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/utils/pointer"
+	aws "sigs.k8s.io/cluster-api-provider-aws/pkg/apis/awsprovider/v1beta1"
+	yaml "sigs.k8s.io/yaml"
+)
+
+func TestValidateAWSProviderSpec(t *testing.T) {
+
+	testCases := []struct {
+		testCase      string
+		machine       *Machine
+		providerSpec  *aws.AWSMachineProviderConfig
+		expectedError string
+		expectedOk    bool
+	}{
+		{
+			testCase: "with no ami values it fails",
+			providerSpec: &aws.AWSMachineProviderConfig{
+				AMI:          aws.AWSResourceReference{},
+				InstanceType: "m4.large",
+				IAMInstanceProfile: &aws.AWSResourceReference{
+					ID: pointer.StringPtr("profileID"),
+				},
+				UserDataSecret: &corev1.LocalObjectReference{
+					Name: "secret",
+				},
+				CredentialsSecret: &corev1.LocalObjectReference{
+					Name: "secret",
+				},
+				SecurityGroups: []aws.AWSResourceReference{
+					{
+						ID: pointer.StringPtr("sg"),
+					},
+				},
+				Subnet: aws.AWSResourceReference{
+					ID: pointer.StringPtr("subnet"),
+				},
+			},
+			expectedOk:    false,
+			expectedError: "providerSpec.ami: Required value: expected either providerSpec.ami.arn or providerSpec.ami.filters or providerSpec.ami.id to be populated",
+		},
+		{
+			testCase: "with no ami values and no instanceType it fails",
+			providerSpec: &aws.AWSMachineProviderConfig{
+				AMI:          aws.AWSResourceReference{},
+				InstanceType: "",
+				IAMInstanceProfile: &aws.AWSResourceReference{
+					ID: pointer.StringPtr("profileID"),
+				},
+				UserDataSecret: &corev1.LocalObjectReference{
+					Name: "secret",
+				},
+				CredentialsSecret: &corev1.LocalObjectReference{
+					Name: "secret",
+				},
+				SecurityGroups: []aws.AWSResourceReference{
+					{
+						ID: pointer.StringPtr("sg"),
+					},
+				},
+				Subnet: aws.AWSResourceReference{
+					ID: pointer.StringPtr("subnet"),
+				},
+			},
+			expectedOk:    false,
+			expectedError: "[providerSpec.ami: Required value: expected either providerSpec.ami.arn or providerSpec.ami.filters or providerSpec.ami.id to be populated, providerSpec.instanceType: Required value: expected providerSpec.instanceType to be populated]",
+		},
+		{
+			testCase: "with all required values it succeed",
```

Comment:
Nit, we have `fails` at the end of others, so this should be `succeeds`
```suggestion
			testCase: "with all required values it succeeds",
```